src/Core/WinSWCore/Configuration/YamlConfiguration.cs

```diff
@@ -266,5 +194,95 @@ public class YAMLLog : Log
         // TODO
         XmlNode? IWinSWConfiguration.ExtensionsConfiguration => throw new NotImplementedException();
 
+        public class YAML_SC_ACTION
+        {
+            [YamlMember(Alias = "type")]
+            private SC_ACTION_TYPE type;
+
+            [YamlMember(Alias = "delay")]
+            private TimeSpan delay;
+
+            public SC_ACTION_TYPE Type { get => type; set => type = value; }
+            public TimeSpan Delay { get => delay; set => delay = value; }
+        }
+
+        public bool HasServiceAccount()
+        {
+            return !(ServiceAccount is null);
+        }
+
+        private string GetArguments(string args, ArgType type)
+        {
+
+            if (args is null)
+            {
+                switch (type)
+                {
+                    case ArgType.arg:
+                        return Defaults.Arguments;
+                    case ArgType.startarg:
+                        return Defaults.StartArguments;
+                    case ArgType.stoparg:
+                        return Defaults.StopArguments;
+                }
+            }
+
+            string newArgs = Regex.Replace(args, @"\\n", " ");
+            return newArgs;
+        }
+
+        private enum ArgType
+        {
+            arg = 0,
+            startarg = 1,
+            stoparg = 2
+        }
+
+
+        public string Id => string.IsNullOrEmpty(_Id) ? Defaults.Id : _Id;
+
+        public string Description => string.IsNullOrEmpty(_Description) ? Defaults.Description : _Description;
+
+        public string Executable => string.IsNullOrEmpty(_Executable) ? Defaults.Executable : _Executable;
+        
+        public string ExecutablePath => string.IsNullOrEmpty(_ExecutablePath) ? Defaults.ExecutablePath : _ExecutablePath;
+
+        public string Caption => string.IsNullOrEmpty(_Caption) ? Defaults.Caption : _Caption;
+
+        public string Arguments => GetArguments(Arguments, ArgType.arg);
+
+        public string? StartArguments => GetArguments(StartArguments, ArgType.startarg);
+
+        public string? StopArguments => GetArguments(StopArguments, ArgType.stoparg);
+
+        public SC_ACTION[] FailureActions
+        {
+            get
+            {
+                var arr = new List<SC_ACTION>();
+
+                foreach (var item in YamlFailureActions)
+                {
+                    arr.Add(new SC_ACTION(item.Type, item.Delay));
+                }
+
+                return arr.ToArray();
+            }
+        }
+
+        public TimeSpan ResetFailureAfter => _ResetFailureAfter.Equals(TimeSpan.Zero) ? Defaults.ResetFailureAfter : _ResetFailureAfter;
+
+        public string WorkingDirectory => string.IsNullOrEmpty(_WorkingDirectory) ? Defaults.WorkingDirectory : _WorkingDirectory;
+
+        public ProcessPriorityClass Priority => _Priority is 0 ? Defaults.Priority : _Priority;
```

Comment:
💡 Use `Nullable<T>`, i.e. `T?` for optional value type members.

Reply:
Should it be even though a not null value set from Defaults?

Reply:
I commented in the wrong place. Was referring to the underlying `[YamlMember]` properties.